vignettes/markdown.Rmd

```diff
@@ -1,17 +1,188 @@
-<!--
-%\VignetteEngine{knitr::knitr}
-%\VignetteIndexEntry{Write R documentation in Markdown}
--->
+---
+title: "Write R Documentation in Markdown"
+author: "Gábor Csárdi"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Write R Documentation in Markdown}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
 
-```{r, echo = FALSE, message = FALSE}
-knitr::opts_chunk$set(
-  comment = "#>",
-  error = FALSE,
-  tidy = FALSE
-)
+```{r, include = FALSE}
+knitr::opts_chunk$set(comment = "#>", collapse = TRUE)
 ```
 
-# Write R documentation in Markdown
+# Introduction
 
-This is a vignette to be written once we know what exactly the Markdown
-support will cover.
+Starting from version 6.0.0, roxygen supports markdown markup within most roxygen tags. Roxygen uses the `commonmark` package, which is based on the CommonMark Reference Implementation to parse these tags. See http://commonmark.org/help/ for more about the parser and the markdown language it supports.
+
+# Turning on markdown support
+
+Currently you have to turn on markdown formatting manually for each roxygen chunk. You can do this by using the `@md` tag anywhere in the chunk. In the future roxygen will provide a global switch that turns on markdown formatting for the whole package. (And also a `@noMd` to turn it off, optionally.)
+
+Here is an example roxygen chunk that uses markdown.
+
+```r
+#' Use roxygen to document a package.
+#'
+#' This function is a wrapper for the [](roxygen2::roxygenize) function from 
```

Comment:
Hmmm, this is a good idea. I agree that the current syntax is awkward, but it was the best we could come up with. THere is a long discussion about it here: https://github.com/gaborcsardi/maxygen/issues/3

commonmark supports autolink, but the contents are restricted:
```r
❯ x <- markdown_xml(text = "this is an <http://roxygen2::roxygenize> autolink")
❯ cat(x)
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">
<document xmlns="http://commonmark.org/xml/1.0">
  <paragraph>
    <text>this is an </text>
    <link destination="http://roxygen2::roxygenize" title="">
      <text>http://roxygen2::roxygenize</text>
    </link>
    <text> autolink</text>
  </paragraph>
</document>
❯ x <- markdown_xml(text = "this is an <roxygen2::roxygenize> autolink")
❯ cat(x)
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">
<document xmlns="http://commonmark.org/xml/1.0">
  <paragraph>
    <text>this is an </text>
    <text>&lt;</text>
    <text>roxygen2::roxygenize&gt; autolink</text>
  </paragraph>
</document>
```

So in practice we cannot really use this to create links. It is a bummer, I agree, and any more ideas are welcome.

In theory we can modify the parser as well, it is Jeroen's package. :) But that seems like a slippery slope.....